extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -189,18 +200,19 @@ private AuthenticationMechanism getAuthenticationMechanism(String authMechanism)
         return mechanism;
     }
 
-    private MongoCredential createMongoCredential(CredentialConfig config) {
-        String username = config.username.orElse(null);
+    private MongoCredential createMongoCredential(MongoClientConfig config) {
+        String username = config.credentials.username.orElse(null);
         if (username == null) {
             return null;
         }
 
-        char[] password = config.password.map(String::toCharArray).orElse(null);
-        //admin is the default auth source in mongo and null is not allowed
-        String authSource = config.authSource.orElse("admin");
+        char[] password = config.credentials.password.map(String::toCharArray).orElse(null);
+        // get the auhtsource, or the database from the config or admin as it is the default auth source in mongo
```

Comment:
get ‘authsource’ or the database config.
Defaults to ‘admin’. 

Can we make ‘admin’ default value for ‘database’ config item?

Reply:
I prefer not, it's a bit dangerous as the database config will be used in something else (a future Panache Mongo implementation) and it will implies that if a user don't configure explicitly the database config it will default to admin and the data will be store in the admin database !

Reply:
Also there is a typo - should read ‘authsource’. 

I get it now. Thanks for clarifying.

Reply:
@machi1990 I fixed the typo, thanks